tools/lsp/preview/drop_location.rs

```diff
@@ -136,20 +172,28 @@ pub fn drop_at(
         to_insert
     };
 
-    let selection_offset = drop_info.insertion_position.offset()
-        + component_text
-            .chars()
-            .take_while(|c| c.is_whitespace())
-            .map(|c| c.len_utf8())
-            .sum::<usize>() as u32;
+    let mut selection_offset = drop_info.insertion_position.offset()
+        + new_text.chars().take_while(|c| c.is_whitespace()).map(|c| c.len_utf8()).sum::<usize>()
+            as u32;
+
+    let (path, _) = drop_info.target_element_node.path_and_offset();
+
+    let doc = tl.get_document(&path)?;
+    let mut edits = Vec::with_capacity(2);
+    if let Some(edit) = completion::create_import_edit(doc, &component_type, &Some(import_path)) {
+        if let Some(sf) = doc.node.as_ref().map(|n| &n.source_file) {
+            selection_offset = TextOffsetAdjustment::new(&edit, sf).adjust(selection_offset);
+        }
+        edits.push(edit);
+    }
+
+    let source_file = doc.node.as_ref().unwrap().source_file.clone();
+
+    let ip = util::map_position(&source_file, drop_info.insertion_position.offset().into());
+    edits.push(lsp_types::TextEdit { range: lsp_types::Range::new(ip.clone(), ip), new_text });
 
     Some((
-        crate::common::ComponentAddition {
```

Comment:
the `ComponentAddition` struct in `common.rs` seems to be unused now. Should it also be removed?

Reply:
I can still see it in your branch (and it's not in the diff) : https://github.com/hunger/slint/blob/dec50d026bf9bef9fe72b6f9ab914642e9d0241a/tools/lsp/common.rs#L157